test/unit/specs/components/common/SessionApprove.spec.js

```diff
@@ -5,16 +5,12 @@ import SessionApprove from "common/SessionApprove"
 describe(`SessionApprove`, () => {
   const localVue = createLocalVue()
   localVue.use(Vuelidate)
-  localVue.directive("focus", () => {})
 
   let wrapper, $store
 
   beforeEach(() => {
     const getters = {
-      signRequest: {
-        senderAddress: "cosmos1234",
-        signMessage: "{}"
-      }
+      signRequest: `{signMessage:{"account_number":"1","chain_id":"testnet","fee":{"amount":[{"amount":"40","denom":"stake"}],"gas":"39953"},"memo":"(Sent via Lunie)","msgs":[{"type":"cosmos-sdk/MsgSend","value":{"amount":[{"amount":"12000000","denom":"stake"}],"from_address":"cosmos1ek9cd8ewgxg9w5xllq9um0uf4aaxaruvcw4v9e","to_address":"cosmos1324vt5j3wzx0xsc32mjhkrvy5gn5ef2hrwcg29"}}],"sequence":"0"}}`
```

Comment:
This is why your tests are failing. Your computed properties are trying to parse `this.signRequest.signMessage`, which is incorrect. At this point `this.signRequest` is a string, and there is no `signMessage` property.

(The giveaway is in the error here: `SyntaxError: Unexpected token u in JSON at position 0`. The `u` it refers to is `'undefined'`)

My suggestion is to have one computed property which runs a `JSON.parse()` on the request, and then have other computed properties use this result.

I also couldn't find a getter for `signRequest`, but perhaps have that parse the JSON for you?

Reply:
`signRequest` should be parsed when adding it to the store in the extension I think. The getter is only in the extension.